transcom/mymove
# INT B-20793 Add GSR role and base (QAE) traits in office/admin app

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20793)

## Summary

As we continue implementation of creating GSR (Government Surveillance Representative) users, we need to set the base permissions the same as QAE. This has also been requested to be placed on a feature flag, so this PR also implements that logic. Nothing too crazy.

This PR does the following:
- adds all relevant information needed for GSR feature flag 
(you can find the merged PR for `milmove-feature-flags` repo [here](https://github.com/transcom/milmove-feature-flags/pull/196))
- implements creating a GSR role using the local sign-in method
- adds GSR checkbox to admin app for creating/updating/viewing office users
- added in functionality to NOT create an okta account when approving requested office users from devlocal
- set default log in route for GSR & base permissions to be the same as QAE
- checking GSR feature flag on sign in, if `gsr_role` is `false` then that GSR user will get the invalid permissions page
- fixing the requested office users form to not be skinny on load by setting static width
- updated all relevant tests

### How to test

1. In your `envrc` and `developmental-features` files, set the GSR role value to `true`
2. FIre up your `server` and `client`
3. Using local sign-in, create a new GSR user
4. Confirm you are directed to the QAE home page where you can search for whatever
5. Navigate the app, confirm you can do all things QAE
6. Log out
7. Set the flags to `false`
8. Refire your `server`
9. Log in as a GSR
10. You should see the invalid permissions screen

## Screenshots
![Screenshot 2024-08-21 at 10 25 38 AM](https://github.com/user-attachments/assets/417789c7-8fae-40b0-8a26-d71cdb69eca7)

![Screenshot 2024-08-21 at 10 25 13 AM](https://github.com/user-attachments/assets/e7a41f41-66e4-4e6a-80e5-c482670937b2)

508
![Screenshot 2024-08-22 at 10 13 17 AM](https://github.com/user-attachments/assets/cfc32ab8-bc10-4968-a6c6-e0e8a89cb85e)
